browser/widevine/widevine_permission_request.cc

```diff
@@ -33,31 +38,45 @@ WidevinePermissionRequest::WidevinePermissionRequest(
 
 WidevinePermissionRequest::~WidevinePermissionRequest() = default;
 
+#if BUILDFLAG(IS_ANDROID)
+std::u16string WidevinePermissionRequest::GetDialogMessageText() const {
+  return l10n_util::GetStringFUTF16(
+      GetWidevinePermissionRequestTextFrangmentResourceId(false),
+      url_formatter::FormatUrlForSecurityDisplay(
+          requesting_origin(),
+          url_formatter::SchemeDisplay::OMIT_CRYPTOGRAPHIC));
+}
+#else
 std::u16string WidevinePermissionRequest::GetMessageTextFragment() const {
   return brave_l10n::GetLocalizedResourceUTF16String(
       GetWidevinePermissionRequestTextFrangmentResourceId(for_restart_));
 }
+#endif
 
 void WidevinePermissionRequest::PermissionDecided(ContentSetting result,
                                                   bool is_one_time,
                                                   bool is_final_decision) {
   // Permission granted
   if (result == ContentSetting::CONTENT_SETTING_ALLOW) {
-#if BUILDFLAG(IS_LINUX)
-    // Prevent relaunch during the browser test.
-    // This will cause abnormal termination during the test.
-    if (for_restart_ && !is_test_) {
-      // Try relaunch after handling permission grant logics in this turn.
-      base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
-          FROM_HERE, base::BindOnce(&chrome::AttemptRelaunch));
-    }
-#endif
     if (!for_restart_) {
-      EnableWidevineCdmComponent();
+      EnableWidevineCdm();
+    } else {
+#if BUILDFLAG(IS_ANDROID)
+      EnableWidevineCdm();
+#endif
+      // Prevent relaunch during the browser test.
+      // This will cause abnormal termination during the test.
+      if (!is_test_) {
+        // Try relaunch after handling permission grant logics in this turn.
+        base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+            FROM_HERE, base::BindOnce(&chrome::AttemptRelaunch));
+      }
     }
     // Permission denied
   } else if (result == ContentSetting::CONTENT_SETTING_BLOCK) {
-    DontAskWidevineInstall(web_contents_, dont_ask_widevine_install_);
+    Profile* profile =
+        static_cast<Profile*>(web_contents_->GetBrowserContext());
+    permissions::AskWidevineInstall(profile->GetPrefs(), !get_dont_ask_again());
```

Comment:
`get_dont_ask_again` should also be switch to match ask instead of don't ask, but not sure if this is needed per our DM discussion